test/spec_request.rb

```diff
@@ -1188,6 +1217,21 @@ def ip_app
 
     res = mock.get '/',
       'REMOTE_ADDR' => '1.2.3.4',
+      'HTTP_FORWARDED' => 'for=3.4.5.6'
+    res.body.must_equal '1.2.3.4'
+
+    res = mock.get '/',
+      'HTTP_X_FORWARDED_FOR' => '3.4.5.6',
+      'HTTP_FORWARDED' => 'for=5.6.7.8'
+    res.body.must_equal '5.6.7.8'
+
+    res = mock.get '/',
+      'HTTP_X_FORWARDED_FOR' => '3.4.5.6',
+      'HTTP_FORWARDED' => 'for=5.6.7.8, for=7.8.9.0'
+    res.body.must_equal '7.8.9.0'
+
+    res = mock.get '/',
+      'REMOTE_ADDR' => '1.2.3.4',
```

Comment:
Test coverage demonstrating that trusted proxy detection behaves with Forwarded, too?